extensions/interactions/ItemSelectionInput/validator.js

```diff
@@ -89,15 +89,60 @@ oppia.filter('oppiaInteractiveItemSelectionInputValidator', [
         baseInteractionValidationService.getAnswerGroupWarnings(
           answerGroups, stateName));
 
-      if (!defaultOutcome ||
-          $filter('isOutcomeConfusing')(defaultOutcome, stateName)) {
+      var selectedChoices = [];
+      if (maxAllowedCount === 1) {
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
+          for (var j = 0; j < ruleSpecs.length; j++) {
+            for (var k = 0; k < ruleSpecs[j].inputs.x.length; k++) {
+              var choicePreviouslySelected = (
+                selectedChoices.indexOf(ruleSpecs[j].inputs.x[k]) !== -1);
+              if (!choicePreviouslySelected) {
+                selectedChoices.push(ruleSpecs[j].inputs.x[k]);
+              }
+            }
+          }
+        }
+      }
+      if (selectedChoices.length < numChoices) {
+        if (!defaultOutcome ||
+            $filter('isOutcomeConfusing')(defaultOutcome, stateName)) {
+          warningsList.push({
+            type: WARNING_TYPES.ERROR,
+            message: (
+              'Please clarify the default outcome so it is less confusing to ' +
+              'the user.')
+          });
+        }
+      }
+      // For situations where max choices is 1,
+      // ensure that number of answers matches number of choices
+      if (maxAllowedCount === 1 && answerGroups.length > numChoices) {
         warningsList.push({
           type: WARNING_TYPES.ERROR,
           message: (
-            'Please clarify the default outcome so it is less confusing to ' +
-            'the user.')
+            'Please ensure that the number of choices and answers match.')
         });
       }
+      // For situations where max choices is 1,
+      // ensure that containsAtLeastOneOf only handles one option
+      if (maxAllowedCount === 1) {
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
+          for (var j = 0; j < ruleSpecs.length; j++) {
+            for (var k = 0; k < ruleSpecs[j].inputs.x.length; k++) {
+              if (ruleSpecs[j].rule_type === 'ContainsAtLeastOneOf' &&
```

Comment:
Ah! I'm very sorry, this is my fault. The second line was wrong. It should be "contains one of the given choices", and that rule may have any number of items ticked.

My apologies for misleading you -- does this clear things up? To be clear again, this is my fault -- I wrote the wrong thing in my review comment.